packages/flutter_tools/lib/src/build_system/targets/native_assets.dart

```diff
@@ -13,61 +13,64 @@ import '../../dart/package_map.dart';
 import '../../isolated/native_assets/native_assets.dart';
 import '../build_system.dart';
 import '../depfile.dart';
-import '../exceptions.dart';
+import '../exceptions.dart' show MissingDefineException;
 import 'common.dart';
 
+export '../../isolated/native_assets/native_assets.dart'
```

Comment:
This cannot roll into g3. Nothing in the not-isolate directory can depend on `isolated/`.
```suggestion
export '../../isolated/native_assets/native_assets.dart'
```

Reply:
Is there prior art on how to resolve this? Duplicating code, json serde, or something?

Reply:
Maybe I can get away with wiring just some of the fields through. Let me check.

Reply:
Apparently I misremembered, inside `build_system/targets` we can use the types.

> The build_system/targets dir is already excluded in other build systems.

https://github.com/flutter/flutter/pull/143055